### DISCUSSION TITLE
Credit Card Payment Scheduling / Bills

### DISCUSSION THREAD

**Discussion by RegalMonkey** [firefly-iii/firefly-iii#4671] [Firefly III]
I am running Firefly III version 5.5.7.

Hello, 

Just got a fresh install setup and was looking to mimic the functions I'm accustomed to from using KMyMoney. (https://kmymoney.org/) and have run into some issues with the credit cards. I get that they're considered an asset in here, which so long as one remembers to mark it as a negative balance initially works fine. However I'm having issues with the payment schedule setup.

![image](https://user-images.githubusercontent.com/82684917/115094868-bd43c780-9ee4-11eb-8c8d-501a1ef60883.png)

I see the field monthly payment date, but aside from it not showing any other options than full payment, it also doesn't seem to have any bearing on creating a 'bills' item. I do like to check up on each card when it would be due even if there's no balance just to make sure there's nothing on there.

On a related note, for the bills menu. It would be particularly handy to add a 'skip this instance' button to just bump it out by one interval for cases like the above where the payment would be $0 allowing for a check-and-go in a sense.

Any clarification needed, just ask.

Thanks.

**Comment by RegalMonkey** [comment]
Actually, thinking on it further even adding a link into the bills item to create a transaction based on predetermined parameters, from a given asset to a given expense would save a bunch of complication of having to write in rules to associate a transaction with a bill. 

Pressing an enter transaction button could just open the transaction menu with the expected from/to/amounts pre-populated to allow for editing and entry, then the interval is marked fulfilled. Granted I'm no coder but that seems a lot simpler than associating transactions backwards to the bill.

**Comment by JC5** [comment]
> I see the field monthly payment date, but aside from it not showing any other options than full payment, it also doesn't seem to have any bearing on creating a 'bills' item.

No. That's up to you. Firefly III won't provide it.

> On a related note, for the bills menu. It would be particularly handy to add a 'skip this instance' button to just bump it out by one interval for cases like the above where the payment would be $0 allowing for a check-and-go in a sense.

Bills aren't really meant for these things. Personally, I don't create bills for credit cards at all. The payments I make for a credit card are regular withdrawals and paying the balance of the credit card is a transfer.

**Reply by RegalMonkey** [reply]
> Bills aren't really meant for these things. Personally, I don't create bills for credit cards at all. The payments I make for a credit card are regular withdrawals and paying the balance of the credit card is a transfer.

Yes, but those transfers have a due date which would be suitable to call a bill, particularly if the balance is not paid in full at any given time. I note that there's a field for a 'Credit card monthly payment date' when editing the card, so I expect there is some relevance to that field. Is there someplace else where an upcoming CC payment/transfer due is flagged?

The skip-interval button would be a good, and I expect simple to implement, thing as well. Even for regular utility bills it can be of use. For example where I live there is a bizarre rule about the electricity bill having to have a 45 day window for payment that a couple times a year will result in getting one that shows 2 months worth of money due, but not meant to be paid for over a month, so in theory one could skip the current month to pay them both on the next cycle, or pay them both now and skip next month. Having a 'skip interval' button on the bill would make for a easy adjustment of the next due date.

**Reply by JC5** [reply]
> Yes, but those transfers have a due date which would be suitable to call a bill, particularly if the balance is not paid in full at any given time. I note that there's a field for a 'Credit card monthly payment date' when editing the card, so I expect there is some relevance to that field.

The choice I made is not to have those as bills. Although it's absolutely true what you say, Firefly III will not be able to treat these transfers as bills.

> The skip-interval button would be a good, and I expect simple to implement, thing as well.

I'm sure you do. But you can skip bills automatically by setting the "skip" field as you require. A separate button will not be added.
